### ISSUE TITLE
Getting Invalid Date when trying to parse string that was previously formatted with a timezone

### ISSUE THREAD

**Issue by ryan-snyder** [date-fns/date-fns#1438]
Versions:
date-fns: 2.0.0-alpha.27
date-fns-tz: 1.0.7

I'm trying to parse a string that was previously formatted with
```js
const date = format(parseISO(date), 'MMM do yy, h:mm a', {
  timeZone: 'America/Halifax',
});
```

But when I parse it like so:
```js
parseISO(date)
```
it returns Invalid Date.  That seems to indicate that the date is improperly formatted but since I have already formatted it, that doesn't make sense

**Comment by kossnocorp** [comment]
Hi! Please provide the initial date value that you pass to `parseISO`.

**Comment by ryan-snyder** [comment]
Original date is in a format like `2019-09-19T19:12:50.486Z`. That is formatted into a human readable date in my timezones. That all works. But later on, when I need to compare these dates, parseISO returns `Invalid Date`

**Comment by Imballinst** [comment]
@ryan-snyder `parseISO` only allows ISO8601 format, isn't it? https://en.wikipedia.org/wiki/ISO_8601. From your case, it looks to me that:

1. You parse the _correct_ ISO Date
2. Then, you format it to `MMM do yy, h:mm a`
3. Then, you use `parseISO` with the result of (2) -- because `MMM do yy, h:mm a` is not an ISO 8601 format -- probably that's the case?

Let me know is my understanding is wrong.

**Comment by ryan-snyder** [comment]
@Imballinst Yes that does sound correct. From what I can tell from the documentation, there's no way to turn a non ISO formatted date into a date that date-fns can parse/compare, correct? If so I'll just change the format to a valid ISO format.

**Comment by Imballinst** [comment]
@ryan-snyder hmm, I think `parse` still exists for non-ISO format: https://github.com/date-fns/date-fns/blob/master/src/parse/index.js.

I just tried it on https://npm.runkit.com/date-fns, seems working fine.

<img width="703" alt="image" src="https://user-images.githubusercontent.com/7077157/66350902-2c636600-e9a8-11e9-90bb-edc8976ddff8.png">
